### DISCUSSION TITLE
how to do AsDiscrete(argmax=True) on the correct channel?

### DISCUSSION THREAD

**Discussion by lg2578** [Project-MONAI/MONAI#4374] [Q&A] [answered]
when I tried to do ,on a batch data from model's output,I have to do like this:
```
              val_outputs=model(val_images)
                val_outputs=val_outputs.swapaxes(0,1)
                val_outputs=post_trans(val_outputs)
                val_outputs=val_outputs.swapaxes(0,1)
                
                val_labels=val_labels.swapaxes(0,1)
                val_labels=post_label(val_labels)
                val_labels=val_labels.swapaxes(0,1)
                
               dice_metric(y_pred=val_outputs, y=val_labels)
```
because of  AsDiscrete  set to first dim as channel.
is there a way not need to swapaxes？ and the code more elegant
it seems when loadimage,monai keeps the channels info ,

**Comment by Nic-Ma** [comment, marked as answer]
Hi @lg2578 ,

Thanks for your interest here.
The typical method of postprocessing is: `data = [postprocessing(i) for i in decollate_batch(data)]`.
You can find all the details in this tutorial:
https://github.com/Project-MONAI/tutorials/blob/main/modules/decollate_batch.ipynb

Thanks.

**Reply by lg2578** [reply]
thank you very much ,I  found Transposed is what I needed. so that swap axes can be done in a transform compose.
